docs/DevGuide/CompilerLinkerErrors.md

```diff
@@ -0,0 +1,115 @@
+`test_upwind_flux_random()`.
+
+We have now determined what the linker error is (a missing link dependency), and
+in which library, file, and function the missing link dependency is used. We now
+need to understand what the missing link dependency is. Since SpECTRE uses a lot
+of templates, the missing missing reference (link dependency) can be quite
+long. In this case it is:
+```
+Tensor<DataVector,
+brigand::list<brigand::integral_constant<int, 1> >,
+brigand::list<Tensor_detail::TensorIndexType<3ul, (UpLo)0, Frame::Inertial,
+(IndexType)0> > >
+random_unit_normal<DataVector>(gsl::not_null<std::mersenne_twister_engine<
+unsigned long, 32ul, 624ul, 397ul, 31ul, 2567483615ul, 11ul, 4294967295ul, 7ul,
+2636928640ul, 15ul, 4022730752ul, 18ul, 1812433253ul>*>, Tensor<DataVector,
+brigand::list<brigand::integral_constant<int, 1>,
+brigand::integral_constant<int, 1> >,
+brigand::list<Tensor_detail::TensorIndexType<3ul, (UpLo)1, Frame::Inertial,
+(IndexType)0>, Tensor_detail::TensorIndexType<3ul, (UpLo)1, Frame::Inertial,
+(IndexType)0> > > const&)
+```
+In order to make this easier to read, it is recommended to run the code through
+ClangFormat. This can be done by copying the linker output into an empty `cpp`
+file and running `clang-format -i EMPTY_FILE_WITH_LINKER_OUTPUT`. Keep in mind
+that ClangFormat will only work if you have only copied the part of the linker
+output that resembles valid C++. Doing so in this case gives:
+\code{.cpp}
+Tensor<DataVector, brigand::list<brigand::integral_constant<int, 1> >,
+       brigand::list<Tensor_detail::TensorIndexType<
+           3ul, (UpLo)0, Frame::Inertial, (IndexType)0> > >
+random_unit_normal<DataVector>(
+    gsl::not_null<std::mersenne_twister_engine<
+        unsigned long, 32ul, 624ul, 397ul, 31ul, 2567483615ul, 11ul,
+        4294967295ul, 7ul, 2636928640ul, 15ul, 4022730752ul, 18ul,
+        1812433253ul>*>,
+    Tensor<DataVector,
+           brigand::list<brigand::integral_constant<int, 1>,
+                         brigand::integral_constant<int, 1> >,
+           brigand::list<
+               Tensor_detail::TensorIndexType<3ul, (UpLo)1, Frame::Inertial,
+                                              (IndexType)0>,
+               Tensor_detail::TensorIndexType<3ul, (UpLo)1, Frame::Inertial,
+                                              (IndexType)0> > > const&);
+\endcode
+We see that it is the function `random_unit_normal` that isn't found. Now we can
+search in the code base where than function is defined. Doing a
+`git grep "random_unit_normal"` points to
+`tests/Unit/Helpers/DataStructures/RandomUnitNormal.?pp`. Looking
+at `tests/Unit/Helpers/DataStructures/RandomUnitNormal.hpp` we see that the
+function `random_unit_normal` is declared there, and looking in the
+corresponding `cpp` we see `random_unit_normal` is instantiated in the source
+file. Opening up `tests/Unit/Helpers/DataStructures/CMakeLists.txt` we see that
+the library name is `DataStructuresHelpers`. Thus, linking
```

Comment:
name is ..., and `RandomUnitNormal. cpp` is in the list of sources for the library.